### ISSUE TITLE
GM1 Editor first Release

### ISSUE THREAD

**Issue by Gaaammmler** [Sh0wdown/UnofficialCrusaderPatch#349]
Hello Guys,

i realeased the first Verion from my GM1 Editor: https://github.com/Gaaammmler/Gm1KonverterCrossPlatform/releases

Maybe @Sh0wdown can you link the Editor in your Readme, so more People can see it? 

![bild1](https://user-images.githubusercontent.com/5760157/56842483-cff70900-6895-11e9-8a96-f0b8a766be4f.JPG)

![Stronghold_Crusader_Extreme_1](https://user-images.githubusercontent.com/5760157/56842485-d4bbbd00-6895-11e9-9139-825ec954e559.jpg)

Have fun and try it Guys 👍 :)

**Comment by Russianstory** [comment]
Friend! Have you really improved this disgusting program ?? You could?)) Praise you. The old program was very buggy .... it's beautiful !!! Tell us about the functions of the program!
--
Друг! Ты что,  правда улучшил эту отвратительную программу?? Ты смог?)) Хвала тебе. Старая программа очень глючная была....это же прекрасно!!! РАсскажи об функциях программы!

**Comment by Monsterfisch** [comment]
Nice! I assume you are also going to add a feature to export and import bmp files so we can change the data in external programs correct? After all its a bit inefficient to change each color individually. 

@Russianstory from what i can tell atm the main feature of the program is to change the colors used in each of the animation files that make use of a color set, so you could essentially make player one colors pink for example

**Comment by Gaaammmler** [comment]
Thats correct Monsterfisch,

but this function will take a little bit longer, so i made a first Prerelease to change some Colors in the Colortable for new Player Colors etc.

Later i will export all Color Tables + the Images for editing in the upcomming versions

**Comment by Russianstory** [comment]
In this case, I would really like the program developer to listen to me.
There is a big problem: extracting and adding graphics files from the game - from hm1 to bmp and back. There is also a problem in extracting and adding animation. There is also a problem in that it is impossible to extract many files at once and immediately and hold. If you succeed in solving this, it will simplify work with graphics. That is why I have difficulty with the graphic part of my mod))).
--
В таком случае я бы очень хотел, чтоб разработчик программы прислушался ко мне.
Есть большая проблема: извлечение и добавление файлов графики из игры - из гм1 в бмп и обратно. Также есть проблема  в извлечении и добавлении анимации. Ещё есть проблема в том, что невозможно извлечь сразу много файлов и сразу и зажать. сли получится это решить то это упростит работу с графикой. Именно поэтому у меня сложности с графической частью моего мода ))).

**Comment by Monsterfisch** [comment]
sweet definitely looking forward to that day i am dying to finally effectively work on the image files as extracting them 1by1 with the current program is a pain in order to test them in game. in case you are looking for feedback to make the program more userefficient, i am more than open to give my input

**Comment by Gaaammmler** [comment]
Hello Monsterfisch,

that will be nice if u can help me, i added today a new Version for more Userfriendly Support.(I added the export Images options for you so you can try it, the files will be in next realeases got import function.)
https://github.com/Gaaammmler/Gm1KonverterCrossPlatform/releases/tag/0.1a
What do you want me to do next? If u want we can Speak about it, i'm also a German Dude. I only write all in english to help other People understand all of it what i am doing. :)
![newConverterGUI](https://user-images.githubusercontent.com/5760157/56876416-85b58980-6a47-11e9-8ad1-9d5cb829fd05.JPG)

**Comment by Monsterfisch** [comment]
yeah i know you are german i write in english for the same reason. 

You can add me on discord Monsterfish_ #7254 if you want which should make communication there a little easier ;)

As for what to do next I mean being able to extract and import the animated building tiles like gatehouses and towers etc would be pretty nice as it in the current program causes glitchy textures to appear on other textures from the same gm file. 
Another thing that would be fantastic would be if you could add a feature where you can adjust the size of the images for example in my version of the game the roofs for the big square towers are not as wide as the tower itself which limits the range of possibilities of what can be done with it but I can elaborate on that more in private if needed.

**Comment by Gaaammmler** [comment]
To all Guys following the Thread, new Release will comming soon!
I need to do some small bugfixing and allmost all can be done in the Exporter/Importer, i have some Images that show what can be done by the next realease:
![img1](https://user-images.githubusercontent.com/5760157/57491081-b7d2b100-72bb-11e9-9713-a6aa50fe3566.JPG)
![img2](https://user-images.githubusercontent.com/5760157/57491083-b86b4780-72bb-11e9-8202-b826f5c49ed7.JPG)
![img3](https://user-images.githubusercontent.com/5760157/57491084-b86b4780-72bb-11e9-9bb5-f7dc9fd5cc22.JPG)
![img4](https://user-images.githubusercontent.com/5760157/57491085-b86b4780-72bb-11e9-8272-026e789501e3.JPG)
![img5](https://user-images.githubusercontent.com/5760157/57491086-b86b4780-72bb-11e9-827e-474bda7a13a2.JPG)
![img6](https://user-images.githubusercontent.com/5760157/57491087-b903de00-72bb-11e9-9be9-ae6c193c82fd.JPG)

**Comment by cybercorpse** [comment]
You're doing allright! Hope that your work will help all modding community and especially Stronghold Europe team to bring some long-awaited features!

**Comment by Gaaammmler** [comment]
Hello Guys,

i just created a new Prerelease https://github.com/Gaaammmler/Gm1KonverterCrossPlatform/releases/tag/Pre-release

have fun to test it.

**Comment by sdiebel** [comment]
I started to develope a game called Wild Wild West (not that movie with foreigns). I think it will be a browsergame cause there is no Stronghold-1 SDK (i don't like STRG+2 STRG+3 :-b)
Is there a way to export all images? Currently i need to take a screenshot and edit it in gimp.

**Comment by PodeCaradox** [comment]
@sdiebel yes you can do it by click the export button? Or how do you use my converter?

**Comment by sdiebel** [comment]
Thanks man that's what i have needed, i used Stronghold Image Toolbox before and screenshot... but your Converter is perfect for that job. Nice and thank you, maybe i could donate when i reach my goals.

I'll use the images for a game which i'll make private not public and commercial. When the game runs then sure i have to make own graphics as long no license is available. But that's task after finishing the game.

**Comment by PodeCaradox** [comment]
Just have fun with it, and if you want support me just star the project if you want :)

**Comment by sdiebel** [comment]
thanks and I did *gave you a star :-b
